modules/kafka/src/main/java/com/devonfw/module/kafka/common/messaging/trace/impl/MessageSpanExtractor.java

```diff
@@ -0,0 +1,87 @@
+package com.devonfw.module.kafka.common.messaging.trace.impl;
+
+import static brave.internal.HexCodec.lowerHexToUnsignedLong;
+import static com.devonfw.module.kafka.common.messaging.util.MessageUtil.getHeaderValue;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+
+import com.devonfw.module.kafka.common.messaging.impl.LineReader;
+import com.devonfw.module.kafka.common.messaging.impl.MessageParserImpl;
+
+import brave.propagation.TraceContext;
+import brave.propagation.TraceContext.Builder;
+import brave.propagation.TraceContext.Extractor;
+import brave.propagation.TraceContextOrSamplingFlags;
+
+/**
+ * @author ravicm
+ *
+ */
+public class MessageSpanExtractor implements Extractor<ConsumerRecord<String, String>> {
```

Comment:
Devon does not use b3 headers currently but a simple correlation ID (see com.devonfw.module.logging.common.api.LoggingConstants). Please align the implementation with that.

Reply:
for this am awaiting for the decision to remove the Tracer implementation or not .

Reply:
we have discussed in last architect meeting to keep tracer implementation.